### DISCUSSION TITLE
Spamming the escape key should close the osu client immediately.

### DISCUSSION THREAD

**Discussion by andrewhong04** [ppy/osu#19489] [Ideas] [answered]
Just like how you press the X button on the osu window, the osu logo on the main menu fades out, but if you click on it again, it closes immediately. I think the escape button should mirror this feature.

**Comment by peppy** [comment, marked as answer]
If you mean the fact there's currently a confirm dialog: that will eventually only show when there's an ongoing operation.

**Comment by frenzibyte** [comment]
Let me clarify, you're requesting the following behaviour to happen:

https://user-images.githubusercontent.com/22781491/182021467-32fe5a93-e171-4b14-a70c-4126b4c73001.mp4

But instead of clicking on the "close" button, you want it to happen when pressing the <kbd>esc</kbd> key instead? Not really sure why you would ever want to do that.

**Reply by andrewhong04** [reply]
Yup, this is what I meant. Since spamming the X button on the window is already an option, I think it'll be consistent to add that same functionality to the escape button as well.

**Reply by peppy** [reply]
it will never be the same as they are different UX expectations. use alt-f4 if you want a quicker exit, please.
